<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Footnotes

In [1]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

In [2]:
print = 'PRINT'

In [3]:
print('Hello World')

TypeError: 'str' object is not callable

In [1]:
print('Hello World')

Hello World


In [3]:
squares = [i**2 for i in range(10)]
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [4]:
squares = list()
for i in range(10):
    squares.append(i**2)

print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [5]:
%%timeit
squares = [i**2 for i in range(10)]

502 ns ± 29.6 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


502 nanoseconds is the average time the code takes to run for one loop, $\pm$29.6 nanoseconds is the mean standard deviation of 7 runs. Each run is 1000000 loops, which means the code has been run 7000000 times in total.

In [6]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

615 ns ± 18.9 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


615 nanoseconds is the average time the code takes to run for one loop, $\pm$18.9 nanoseconds is the mean standard deviation of 7 runs. Each run is 1000000 loops, which means the code has been run 7000000 times in total.